File: ShopAppSpring/src/main/java/com/duchung/shopappspring/dtos/UserDTO.java
package com.duchung.shopappspring.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.sql.Date;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {

    @NotEmpty(message = "Full name can't be empty")
    @Size(min = 3, max = 30, message = "Full name must be between 3 to 30")
    @JsonProperty("full_name")
    private String fullName;

    @NotEmpty(message = "Phone number can't be empty")
    @JsonProperty("phone_number")
    @Size(min = 1, max = 10, message = "Full name must be between 1 to 10")
    private String phoneNumber;

    @NotEmpty(message = "Address can't be empty")
    private String address;

    @NotEmpty(message = "Password can't be empty")
    private String password;

    @NotEmpty(message = "RePassword can't be empty")
    @JsonProperty("retype_password")
    private String retypePassword;

    @JsonProperty("date_of_birth")
    private Date dateOfBirth;

    @JsonProperty("facebook_account_id")
    private int facebookAccountId;

    @JsonProperty("google_account_id")
    private int googleAccountId;

    @JsonProperty("sex")
    private int sex;

    @JsonProperty("cic")
    private String cic;

    @JsonProperty("email")
    private String email;

    @JsonProperty("role_id")
    private Long roleId;
}

File: shopapp-angular/src/app/admin/dtos/user.dto.ts
export class UserDTO {
  id: number;
  fullname: string;
  phone_number: string;
  address: string;
  is_active: number;
  date_of_birth: Date;
  avatar: string;
  sex: string;
  cic: string;
  email: string;
  role_id: number;

  constructor(data: any) {
    this.id = data.id;
    this.fullname = data.fullname;
    this.phone_number = data.phone_number;
    this.address = data.address;
    this.is_active = data.is_active;
    this.date_of_birth = data.date_of_birth;
    this.avatar = data.avatar;
    this.role_id = data.role_id;
    this.sex = data.sex;
    this.cic = data.cic;
    this.email = data.email;
  }
}
